core/commands/stat_provide.go

```diff
@@ -46,6 +48,17 @@ This interface is not stable and may change from release to release.
 
 		return nil
 	},
-	Encoders: cmds.EncoderMap{},
-	Type:     batched.BatchedProviderStats{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, s *batched.BatchedProviderStats) error {
+			wtr := tabwriter.NewWriter(w, 0, 0, 1, ' ', 0)
+			defer wtr.Flush()
+
+			fmt.Fprintf(wtr, "TotalProvides: %d\n", s.TotalProvides)
+			fmt.Fprintf(wtr, "AvgProvideDuration: %v\n", s.AvgProvideDuration)
```

Comment:
We could use a custom time formatter that restricts the number of significant digits (e.g. no need for mentioning seconds if we're talking about hours and no need for 6 decimal places in the number of milliseconds)

Reply:
Would  [Duration.String() ](https://golang.org/pkg/time/#Duration.String) be enough?

```suggestion
			fmt.Fprintf(wtr, "AvgProvideDuration: %s\n", s.AvgProvideDuration)
```

iirc it will serialize to `1h2m0.3s` string with pretty sane defaults:

> Leading zero units are omitted. As a special case, durations less than one second format use a smaller unit (milli-, micro-, or nanoseconds) to ensure that the leading digit is non-zero. The zero duration formats as 0s. 
> https://golang.org/pkg/time/#Duration.String

Reply:
> https://golang.org/pkg/time/#Duration.String

`%v` should be calling `.String()`. In any event Duration.String still doesn't round it just removes 0s, for example https://play.golang.org/p/SV6mPITSsgb.